### PR TITLE
tool for checking extension build config

### PR DESCRIPTION
**Pull request by zirain**
this can help for issue like https://github.com/istio/istio/issues/47346

### CODE REVIEW

**Review by lei-tang — COMMENTED** [review]
Thanks for adding this tool! Can you add a user guide for this tool?

**Comment by zirain** [comment]
@lei-tang update README

File: tools/extension-check/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+This is a simple tool to check all core extensions from envoy are enabled in [istio/proxy](https://github.com/istio/proxy).
+
+## Usage
+
+```bash
+go run tools/extension-check/main.go \
+  --ignore-extensions "<WELLKNOWN_IGNORE_EXTENSIONS>" \
+  --envoy-extensions-build-config "<ENVOY_EXTENSIONS_BUILD_CONFIG>" \
+  --proxy-extensions-build-config "<PROXY_EXTENSIONS_BUILD_CONFIG>"
+```
```
Comment (lei-tang) [inline tools/extension-check/README.md:12]: Can you add an example command?

Comment (zirain) [reply tools/extension-check/README.md:12]: there's one in `check-style.sh`

Comment (lei-tang) [reply tools/extension-check/README.md:12]: The one in check-style.sh is the same as the one above. An example with concrete values for WELLKNOWN_IGNORE_EXTENSIONS, ENVOY_EXTENSIONS_BUILD_CONFIG and PROXY_EXTENSIONS_BUILD_CONFIG will make it easer for an user. An user should not have to look into the code to figure out example values for ENVOY_EXTENSIONS_BUILD_CONFIG and PROXY_EXTENSIONS_BUILD_CONFIG. 

go run tools/extension-check/main.go \
  --ignore-extensions tools/extension-check/wellknown-extensions \
  --envoy-extensions-build-config "${ENVOY_EXTENSIONS_BUILD_CONFIG}" \
  --proxy-extensions-build-config "${PROXY_EXTENSIONS_BUILD_CONFIG}"

Comment (zirain) [reply tools/extension-check/README.md:12]: I don't expect anyone use this out of istio, but I will try to add more examples.
There's a hack for proxy.

Comment (zirain) [reply tools/extension-check/README.md]: updated

**Review by lei-tang — APPROVED** [review]
Thank you for the tool!

**Pull request merged**